src/main/java/org/broadinstitute/hellbender/utils/read/GoogleGenomicsReadToGATKReadAdapter.java

```diff
@@ -0,0 +1,664 @@
+    }
+
+    @Override
+    public byte[] getBaseQualities() {
+        final List<Integer> baseQualities = genomicsRead.getAlignedQuality();
+        if ( baseQualities == null || baseQualities.isEmpty() ) {
+            return new byte[0];
+        }
+
+        byte[] convertedBaseQualities = new byte[baseQualities.size()];
+        for ( int i = 0; i < baseQualities.size(); ++i ) {
+            if ( baseQualities.get(i) < 0 || baseQualities.get(i) > Byte.MAX_VALUE ) {
+                throw new GATKException("Base quality score " + baseQualities.get(i) + " is invalid and/or not convertible to byte");
+            }
+            convertedBaseQualities[i] = baseQualities.get(i).byteValue();
+        }
+
+        return convertedBaseQualities;
+    }
+
+    @Override
+    public Cigar getCigar() {
+        if ( genomicsRead.getAlignment() == null || genomicsRead.getAlignment().getCigar() == null ) {
+            return new Cigar();
+        }
+
+        return CigarConversionUtils.convertCigarUnitListToSAMCigar(genomicsRead.getAlignment().getCigar());
+    }
+
+    @Override
+    public String getReadGroup() {
+        // May be null
+        return genomicsRead.getReadGroupId();
+    }
+
+    @Override
+    public int getNumberOfReadsInFragment() {
+        return genomicsRead.getNumberReads() != null ? genomicsRead.getNumberReads() : 0;
+    }
+
+    @Override
+    public int getReadNumber() {
+        // Convert from 0-based to 1-based numbering
+        return genomicsRead.getReadNumber() != null ? genomicsRead.getReadNumber() + 1 : 0;
+    }
+
+    @Override
+    public boolean isPaired() {
+        assertFieldValueNotNull(genomicsRead.getNumberReads(), "number of reads");
+        return genomicsRead.getNumberReads() == 2;
+    }
+
+    @Override
+    public boolean isProperlyPaired() {
+        assertFieldValueNotNull(genomicsRead.getProperPlacement(), "proper placement");
+        return isPaired() && genomicsRead.getProperPlacement();
+    }
+
+    @Override
+    public boolean isUnmapped() {
+        return genomicsRead.getAlignment() == null ||
+               positionIsUnmapped(genomicsRead.getAlignment().getPosition());
+    }
+
+    @Override
+    public boolean mateIsUnmapped() {
+        if ( ! isPaired() ) {
+            throw new IllegalStateException("Cannot get mate information for an unpaired read");
+        }
+
+        return positionIsUnmapped(genomicsRead.getNextMatePosition());
+    }
+
+    private boolean positionIsUnmapped( final Position position ) {
+        return position == null ||
+               position.getReferenceName() == null || position.getReferenceName().equals(SAMRecord.NO_ALIGNMENT_REFERENCE_NAME) ||
+               position.getPosition() == null || position.getPosition() < 0;
+    }
+
+    @Override
+    public boolean isReverseStrand() {
+        assertHasPosition();
+        return assertFieldValueNotNull(genomicsRead.getAlignment().getPosition().getReverseStrand(), "strand");
```

Comment:
this reads weird. `return assert...`. Prefer to split assertion and return to two lines